src/index.spec.ts

```diff
@@ -5,16 +5,36 @@ import semver = require('semver')
 import ts = require('typescript')
 import proxyquire = require('proxyquire')
 import { register, create, VERSION } from './index'
+import { mkdtempSync, readdirSync, copyFileSync, rmdirSync, unlinkSync, existsSync, readFileSync, writeFileSync } from 'fs'
+import * as promisify from 'util.promisify'
 
+const execP = promisify(exec)
+
+const ROOT_DIR = join(__dirname, '..')
 const TEST_DIR = join(__dirname, '../tests')
+const TARBALL_PATH = join(TEST_DIR, 'ts-node-packed.tgz')
 const PROJECT = join(TEST_DIR, 'tsconfig.json')
-const BIN_PATH = join(__dirname, '../dist/bin')
-const BIN_SCRIPT_PATH = join(__dirname, '../dist/bin-script')
+const BIN_PATH = join(TEST_DIR, 'node_modules/.bin/ts-node')
+const BIN_SCRIPT_PATH = join(TEST_DIR, 'node_modules/.bin/ts-node-script')
 
 const SOURCE_MAP_REGEXP = /\/\/# sourceMappingURL=data:application\/json;charset=utf\-8;base64,[\w\+]+=*$/
 
+// Pack and install ts-node locally, necessary to test package "exports"
+before(async function () {
+  this.timeout(30000)
+  const tempDir = mkdtempSync(join(TEST_DIR, 'tmp'))
+  await execP(`npm pack --ignore-scripts "${ROOT_DIR}"`, { cwd: tempDir })
+  const tarballPath = join(tempDir, readdirSync(tempDir)[0])
+  writeFileSync(TARBALL_PATH, readFileSync(tarballPath))
+  unlinkSync(tarballPath)
+  rmdirSync(tempDir)
+  await execP(`npm install`, { cwd: TEST_DIR })
+  const packageLockPath = join(TEST_DIR, 'package-lock.json')
+  existsSync(packageLockPath) && unlinkSync(packageLockPath)
+})
```

Comment:
I wonder if these a way to do this outside of the test suite or possibly a separate set of tests? Either way this looks like a great start at coverage.

Reply:
I can definitely move the `npm pack` into our `npm build`.  That would be better since it would `npm pack` using the same node and npm versions as we use to `publish`.

Are you thinking you want 2x sets of tests, one unit tests, the other doing these e2e-style tests against an `npm install`?

I thought about that but I wasn't sure how useful it'd be since the tests mostly look e2e-style already.  Your call though, happy to split things out.

Reply:
Let's keep it how it is then, I don't have strong opinions either way but it's great to know you thought about this.